apps/setting/settings.js

```diff
@@ -1036,35 +1036,41 @@ function showAltitude() {
     menuAltitude.value = Math.round(pressure.altitude);
     m.draw();
   }
+  function altitudeDone() {
+    settings.seaLevelPressure = seaLevelPressure;
+    updateSettings();
+  }
+
   Bangle.setBarometerPower(1,"settings");
   Bangle.on("pressure",onPressure);
+  E.on("kill", altitudeDone);
   var seaLevelPressure = Bangle.getOptions().seaLevelPressure;
   if (!isFinite(seaLevelPressure)) seaLevelPressure=1013.25;
   var menuPressure = {value:"-"};
   var menuAltitude = {value:"-"};
   var m = E.showMenu({ "" : {title:/*LANG*/"Altitude",back:() => {
       Bangle.setBarometerPower(0,"settings");
       Bangle.removeListener("pressure",onPressure);
-      settings.seaLevelPressure = seaLevelPressure;
-      updateSettings();
+      E.removeListener("kill",altitudeDone);
+      altitudeDone();
       popMenu(systemMenu());
     }},
     /*LANG*/"Pressure (hPa)" : menuPressure,
     /*LANG*/"Altitude (m)" : menuAltitude,
     /*LANG*/"Adjust up" : function() {
       Bangle.buzz(80);
       seaLevelPressure++;
-      Bangle.setOptions({seaLevelPressure:seaLevelPressure});
+      Bangle.setOptions({seaLevelPressure});
```

Comment:
Is it a typo dropping the key:value pairing?

Reply:
Ah this is [shorthand](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#property_definitions) - `{ x: x }` is equivalent to `{ x }`

Reply:
Whoa - trippy!